modules/UI/welcome_page/WelcomePage.js

```diff
@@ -1,98 +1,32 @@
-/* global $, interfaceConfig, APP */
+/* global $ */
 
-import { generateRoomWithoutSeparator } from '../../util/RoomnameGenerator';
-import UIUtil from '../util/UIUtil';
+function enterRoom() {
+    const $enterRoomField = $("#enter_room_field");
 
-var animateTimeout, updateTimeout;
-
-function enter_room() {
-    var val = $("#enter_room_field").val();
+    var val = $enterRoomField.val();
     if(!val) {
-        val = $("#enter_room_field").attr("room_name");
+        val = $enterRoomField.data("room-name");
     }
     if (val) {
         window.location.pathname = "/" + val;
     }
 }
 
-function animate(word) {
-    var currentVal = $("#enter_room_field").attr("placeholder");
-    $("#enter_room_field").attr("placeholder", currentVal + word.substr(0, 1));
-    animateTimeout = setTimeout(function() {
-        animate(word.substring(1, word.length));
-    }, 70);
-}
-
-function update_roomname() {
-    var word = generateRoomWithoutSeparator();
-    $("#enter_room_field").attr("room_name", word);
-    $("#enter_room_field").attr("placeholder", "");
-    clearTimeout(animateTimeout);
-    animate(word);
-    updateTimeout = setTimeout(update_roomname, 10000);
-}
-
 function setupWelcomePage() {
-    $("#videoconference_page").hide();
-    $("#domain_name").text(
-            window.location.protocol + "//" + window.location.host + "/");
-    if (interfaceConfig.SHOW_JITSI_WATERMARK) {
-        var leftWatermarkDiv =
-            $("#welcome_page_header div[class='watermark leftwatermark']");
-        if(leftWatermarkDiv && leftWatermarkDiv.length > 0) {
-            leftWatermarkDiv.css({display: 'block'});
-            UIUtil.setLinkHref(
-                leftWatermarkDiv.parent(),
-                interfaceConfig.JITSI_WATERMARK_LINK);
-        }
-    }
-
-    if (interfaceConfig.SHOW_BRAND_WATERMARK) {
-        var rightWatermarkDiv =
-            $("#welcome_page_header div[class='watermark rightwatermark']");
-        if(rightWatermarkDiv && rightWatermarkDiv.length > 0) {
-            rightWatermarkDiv.css({display: 'block'});
-            UIUtil.setLinkHref(
-                rightWatermarkDiv.parent(),
-                interfaceConfig.BRAND_WATERMARK_LINK);
-            rightWatermarkDiv.get(0).style.backgroundImage =
-                "url(images/rightwatermark.png)";
-        }
-    }
-
-    if (interfaceConfig.SHOW_POWERED_BY) {
-        $("#welcome_page_header>a[class='poweredby']")
-            .css({display: 'block'});
-    }
+    /*
+    * XXX: We left only going to conference page here because transitions via
+    * React Router isn't implemented yet.
+    */
```

Comment:
We've implemented transitions via React Router already. And Also this file is not being used, so we can remove it from repo.